test/metabase/api/actions_test.clj

```diff
@@ -79,10 +83,14 @@
             (is (re= #"Value does not match schema:.*"
                      (:message (mt/user-http-request :crowberto :post 400 action (dissoc request-body k)))))))))))
 
-(deftest unknown-row-action-gives-404
+(deftest row-delete-action-gives-400-when-matching-more-than-one
   (mt/with-temporary-setting-values [experimental-enable-actions true]
     (mt/with-temp-vals-in-db Database (mt/id) {:settings {:database-enable-actions true}}
-      (mt/user-http-request :crowberto :post 404 "actions/row/fake" (mt/mbql-query venues {:filter [:= $id 1]})))))
+      (let [query-that-returns-more-than-one (mt/mbql-query venues {:filter [:> $id -10]})]
+        (is (< 1 (count (mt/rows (qp/process-query query-that-returns-more-than-one)))))
+        (doseq [{:keys [action]} (mock-requests)]
+          (is (re= #"Sorry, this would effect \d+ rows, but you can only act on 1"
+                   (:message (mt/user-http-request :crowberto :post 400 action query-that-returns-more-than-one)))))))))
```

Comment:
I don't think you should delete the old 404 test. We still want to return a 404 if the action does not exist. Why not keep that old test and add a new one as well.

Reply:
I removed it because unknown row actions is tested in [`four-oh-four-test`](https://github.com/metabase/metabase/pull/22716/files/5093dbd70d0fa94e1129187bdb63f1bee49efebf#diff-df28d4b647ab8f404983cbdd69c74161e5437727184e5091f42d6aabef5f4b69R104).